store/src/java/com/zimbra/soap/ZimbraSoapContext.java

```diff
@@ -346,6 +364,25 @@ else if (format.equals(HeaderConstants.TYPE_JAVASCRIPT))
                     throw ServiceException.AUTH_REQUIRED();
                 }
                 Account account = prov.get(AccountBy.name, value, mAuthToken);
+                // Overriding the value for sendMsgRequest if the account in header is not equal to the from address.
+                // Because in case of Persona(added for the owner's account), the account passed in header is not equal to from address.
+                // if zimbraAllowAnyFromAddress = TRUE for delegate's account then we can create Persona with any email address and the
+                // from address could be different in this case.
+                if (body != null && requestName.equals(MailConstants.SEND_MSG_REQUEST)) {
+                    String fromAddress = null;
+                    SendMsgRequest req = JaxbUtil.elementToJaxb(body, SendMsgRequest.class);
+                    for (EmailAddrInfo addr : req.getMsg().getEmailAddresses()) {
+                        if (addr.getAddressType().equals("f")) {
+                            fromAddress = addr.getAddress();
+                            break;
+                        }
+                    }
```

Comment:
Extract this into a utility method, this has been used multiple times.

Reply:
sure. i'll do it.